ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -530,6 +530,9 @@ public static void preAllocateFileIfNeeded(File file, long capacityBytes)
       throws IOException {
     Runtime runtime = Runtime.getRuntime();
     if (System.getProperty("os.name").toLowerCase().startsWith("linux")) {
+      if (!file.isFile()) {
```

Comment:
Shouldn't you be using `file.exists()` instead of `file.isFile()` here? I understand that `isFile()` does other additional checks for whether the file is a directory and such. But the rest of this method assumes that the file, if it exists, is a normal file (otherwise fallocate would fail, for instance). Using `exists()` makes this clearer.

Could also add javadoc for this method?